src/routes/register.tsx

```diff
@@ -1,67 +1,137 @@
+    }
+    createUserWithEmailAndPassword(auth, email, password)
+      .then((userCredential) => {
+        const user = userCredential.user;
+        console.log(user);
+        set(ref(database, `users/${user.uid}`), {
+          firstName,
+          lastName,
+          phoneNumber,
+        });
+        navigate("/");
+      })
+      .catch((error) => {
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        console.log(errorCode, errorMessage);
+      });
+  };
+  const navigate = useNavigate();
   return (
     <Container component="main" maxWidth="xs" disableGutters>
-      <Stack alignItems="center" spacing={2}>
+      <Stack spacing={2}>
         <Typography variant="h2" align="center" gutterBottom color={"cyan"}>
           Event Planner
         </Typography>
-        <Stack alignItems="center" spacing={2}>
+        <Stack
+          component="form"
+          onSubmit={handleSubmit}
+          alignItems="center"
+          spacing={2}
+        >
           <TextField
             margin="normal"
             label="First Name"
             required
             fullWidth
             autoComplete="given-name"
+            id="firstName"
+            name="firstName"
           />
           <TextField
             margin="normal"
             label="Last Name"
             required
             fullWidth
             autoComplete="family-name"
+            id="lastName"
+            name="lastName"
           />
           <TextField
             margin="normal"
             label="Email"
             required
             fullWidth
-            autoComplete="email"
             type="email"
+            id="email"
+            name="email"
           />
           <TextField
             margin="normal"
             label="Password"
             required
             fullWidth
-            autoComplete="current-password"
             type="password"
+            id="password"
+            name="password"
           />
           <TextField
             margin="normal"
             label="Confirm Password"
             required
             fullWidth
-            autoComplete="current-password"
             type="password"
+            id="confirmPassword"
+            name="confirmPassword"
           />
           <TextField
             margin="normal"
             label="Phone Number"
             fullWidth
-            autoComplete="tel"
             type="tel"
+            id="phoneNumber"
+            name="phoneNumber"
           />
-          <Button
-            variant="contained"
-            color="secondary"
-            type="submit"
-            fullWidth
-            sx={{ mt: 3, mb: 2 }}
-          >
-            Create Account
-          </Button>
+          <Stack direction="row" spacing={2}>
+            <Button
+              variant="contained"
+              color="secondary"
+              type="submit"
+              fullWidth
+              sx={{ mt: 3, mb: 2 }}
```

Comment:
mhm, if you need more spacing between elements inside a stack, just increase the stacks spacing prop